gmso/core/forcefield.py

```diff
@@ -77,6 +82,69 @@ def atom_class_groups(self):
                 atomclass_dict[atom_type.atomclass] = atomclass_group
         return atomclass_dict
 
+    def group_by_expression(self):
+        """Create ForceField member groups having same expressions"""
+        atom_type_groups = self._group_by_expression(_type='AtomType')
+        bond_type_groups = self._group_by_expression(_type='BondType')
+        angle_type_groups = self._group_by_expression(_type='AngleType')
+        dihedral_type_groups = self._group_by_expression(_type='DihedralType')
+        return {
+            'atom_types': atom_type_groups,
+            'bond_types': bond_type_groups,
+            'angle_types': angle_type_groups,
+            'dihedral_types': dihedral_type_groups
+        }
+
+    def _group_by_expression(self, _type):
```

Comment:
Adding developer docs to all methods (including internal methods) would be helpful. I don't know what this function is meant to do. I would also suggest `type_` instead of `_type`, as [described by PEP-8](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles).
> `_single_leading_underscore`: weak "internal use" indicator. E.g. `from M import *` does not import objects whose names start with an underscore.
> `single_trailing_underscore_`: used by convention to avoid conflicts with Python keyword, e.g.
> `Tkinter.Toplevel(master, class_='ClassName')`